pkg/goDB/storage/gpfile/gpfile_test.go

```diff
@@ -181,19 +181,153 @@ func testRoundtrip(t *testing.T, enc encoders.Type) {
 	}
 }
 
+func TestInvalidMetadata(t *testing.T) {
+
+	os.RemoveAll("/tmp/test_db")
+	if err := os.MkdirAll("/tmp/test_db/0", 0755); err != nil {
+		t.Fatalf("Error creating test dir for reading: %s", err)
+	}
+	if err := os.WriteFile("/tmp/test_db/0/.blockmeta", []byte{0x1}, 0644); err != nil {
+		t.Fatalf("Error creating test metdadata for reading: %s", err)
+	}
+
+	testDir := NewDir("/tmp/test_db", 1000, ModeRead)
+	if err := testDir.Open(); err == nil || err.Error() != "error decoding metadata file `/tmp/test_db/0/.blockmeta`: input data too small to be a GPDir metadata header (len: 1)" {
```

Comment:
Something itches when I see direct comparison of error strings. Any chance you can use `errors.Is` here? Possibly, by supplying a sentinel `ErrInputDataTooSmall` and wrap that inside the library?